### PR TITLE
feat(agave): Added Agave to Zapper

### PR DESCRIPTION
**Pull request by liz-cs**
## Checklist

- [ ] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] (optional) As a contributor, my Ethereum address/ENS is:
- [ ] (optional) As a contributor, my Twitter handle is:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,11 +39,11 @@
     "@ethersproject/bytes": "^5.5.0",
     "@ethersproject/contracts": "^5.5.0",
     "@ethersproject/providers": "^5.5.0",
-    "@nestjs/cli": "8.2.4",
+    "@nestjs/cli": "^8.2.6",
```
[the remark this answers is not in the document]
Comment (immasandwich) [reply package.json:42]: Also, restore `.env.sample` from the `main` branch.

File: src/apps/aave-v2/helpers/aave-v2.lending.token-helper.ts [hunk 2]
```diff
@@ -172,11 +172,9 @@ export class AaveV2LendingTokenHelper {
             statsItems,
           },
         };
-
         return token;
       }),
     );
-
```
Comment (immasandwich) [inline src/apps/aave-v2/helpers/aave-v2.lending.token-helper.ts:179]: Remove unnecessary whitespace changes here and in `src/apps/aave-v2/helpers/aave-v2.claimable.contract-position-helper.ts`

File: src/apps/agave/agave.module.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+@Register.AppModule({
+  appId: AGAVE_DEFINITION.id,
+  providers: [
+    AaveV2ClaimableBalanceHelper,
+    AaveV2ClaimableContractPositionHelper,
+    AaveV2ContractFactory,
+    AaveV2HealthFactorMetaHelper,
+    AaveV2LendingBalanceHelper,
+    AaveV2LendingTokenHelper,
```
Comment (immasandwich) [inline src/apps/agave/agave.module.ts:27]: You can simply add `AaveV2AppModule` to the `imports` key instead, since these helper providers are exported from that module.

File: src/apps/agave/gnosis/agave.balance-fetcher.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      groupId: AGAVE_DEFINITION.groups.claimable.id,
+      network,
+    });
+  }
+
+  private async getHealthFactorMeta(address: string) {
+    return this.healthFactorHelper.getHealthFactor({
+      address,
+      network,
+      lendingPoolAddress: '0x5E15d5E33d318dCEd84Bfe3F4EACe07909bE6d9c',
```
Comment (immasandwich) [inline src/apps/agave/gnosis/agave.balance-fetcher.ts:61]: Keep addresses lowercased in code 🙏

File: src/apps/agave/gnosis/agave.stable-borrow.token-fetcher.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    @Inject(AgaveContractFactory) private readonly agaveContractFactory: AgaveContractFactory,
+    @Inject(AaveV2LendingTokenHelper) private readonly aaveV2LendingTokenHelper: AaveV2LendingTokenHelper,
+  ) {}
+
+  async getPositions() {
+    return this.aaveV2LendingTokenHelper.getTokens({
+      appId,
+      groupId,
+      network,
+      protocolDataProviderAddress: '0x24dCbd376Db23e4771375092344f5CbEA3541FC0',
```
Comment (immasandwich) [inline src/apps/agave/gnosis/agave.stable-borrow.token-fetcher.ts:31]: Lowercase lmao

File: src/apps/agave/gnosis/agave.variable-borrow.token-fetcher.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    @Inject(AaveV2LendingTokenHelper)
+    private readonly aaveV2LendingTokenHelper: AaveV2LendingTokenHelper,
+  ) {}
+
+  async getPositions() {
+    return this.aaveV2LendingTokenHelper.getTokens({
+      appId,
+      groupId,
+      network,
+      protocolDataProviderAddress: '0x24dCbd376Db23e4771375092344f5CbEA3541FC0',
```
Comment (immasandwich) [inline src/apps/agave/gnosis/agave.variable-borrow.token-fetcher.ts:32]: lol

File: src/apps/agave/gnosis/agave.stable-borrow.token-fetcher.ts (hunk 5 above)
Comment (liz-cs) [reply src/apps/agave/gnosis/agave.stable-borrow.token-fetcher.ts:31]: got it. I'll check all the addresses

**Review by immasandwich — APPROVED** [review]
